projects/RabbitMQ.Client/client/api/ICredentialsRefresher.cs

```diff
@@ -73,55 +72,86 @@ public class TimerBasedCredentialRefresherEventSource : EventSource
 
     public class TimerBasedCredentialRefresher : ICredentialsRefresher
     {
-        private readonly IDictionary<ICredentialsProvider, TimerRegistration> _registrations =
-            new Dictionary<ICredentialsProvider, TimerRegistration>();
-        private readonly object _lockObj = new();
+        private ICredentialsProvider? _credentialsProvider;
+        private TimerRegistration? _registration;
+        private bool _disposed = false;
 
         public ICredentialsProvider Register(ICredentialsProvider provider, ICredentialsRefresher.NotifyCredentialRefreshedAsync callback)
         {
-            if (!provider.ValidUntil.HasValue || provider.ValidUntil.Value.Equals(TimeSpan.Zero))
+            // TODO should this be ArgumentException?
```

Comment:
From my perspective this should result in an ArgumentException. The argument was clearly null or has some invalid value, returning the provider makes the caller fool into thinking the call and registration was successful.